client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/DefaultNodeProcessor.java

```diff
@@ -45,7 +45,9 @@ public void process(TraceContext context, APMSpan span, NodeBuilder nodeBuilder)
                     } catch (MalformedURLException e) {
                         nodeBuilder.setUri(entry.getValue().toString());
                     }
-                    nodeBuilder.setEndpointType(entry.getKey().substring(0, entry.getKey().length() - 4));
+                    String type=entry.getKey().substring(0, entry.getKey().length() - 4);
```

Comment:
there is already a check for key ending ".url" or ".uri" - so would always be atleast 4